### DISCUSSION TITLE
How to fit long legend string in visible area under plot (line breaks)?

### DISCUSSION THREAD

**Discussion by oddfrost** [JetBrains/lets-plot-kotlin#167] [Q&A] [answered]
Hello.
I'm trying to draw some data from prometheus but having a problem formatting legend.
1. There are no line breaks for long legend string. guideLegend(nrow) creates new line for each entity but inside this entity there are no line breaks.
2. When using ggsave to save plot as html bottom lines are out of page and sight (although it looks fine in Jupyter notebook)

Any ideas how fix it?

- [lets-plot-legend.ipynb.zip](https://github.com/JetBrains/lets-plot-kotlin/files/11027267/lets-plot-legend.ipynb.zip) notebook with example (not tested, some problems with kernel)
- [test.html.zip](https://github.com/JetBrains/lets-plot-kotlin/files/11027277/test.html.zip) html plot example

![test](https://user-images.githubusercontent.com/31819061/226581736-0798f554-c1bf-4b1e-ac78-6148668ee156.svg)

**Comment by alshan** [comment, marked as answer]
> When using ggsave to save plot as html bottom lines are out of page

This is bug in HTML export - I've created an issue : https://github.com/JetBrains/lets-plot/issues/725

> There are no line breaks for long legend string. guideLegend(nrow) creates new line for each entity but inside this entity there are no line breaks.

Technically not a bug (i.e. , expected).
Try to customize labels via `limits, labels` parameters in `scaleColorDiscrete()`, by inserting `\n` char as required.

However, a kind of automatic labels text wrap  might be a useful improvement.

**Reply by oddfrost** [reply]
Although i've marked it as an answer, `scaleColorDiscrete(labels = listOf("a\nb\nc"))` is not working and there are no line breaks (\n\r too).
And there is no break template in [formats](https://github.com/JetBrains/lets-plot-kotlin/blob/master/docs/formats.md) description.

**Reply by alshan** [reply]
You are right, I've added an issue: https://github.com/JetBrains/lets-plot/issues/726

**Reply by alshan** [reply]
Hi, both issues (truncated labels at the bottom, line-breaks in legend labels) are solved in [v3.2.0](https://github.com/JetBrains/lets-plot/releases/tag/v3.2.0). Thanx for the feedback!

**Reply by alshan** [reply]
More precisely in [v4.4.0](https://github.com/JetBrains/lets-plot-kotlin/releases/tag/v4.4.0).
